v2: Placeholders for measurement-specific output parameters (#625)
PEtab-dev/PEtab: doc/v2/documentation_data_format.rst

  clearer. The ``log10`` transformation has been removed, since this was mostly
  relevant for visualization purposes, and the same effect can be achieved by
  rescaling the parameters of the respective (natural) log-distributions.
* The ``observableFormula`` field in the :ref:`v2_observables_table` must not
  contain any observable IDs. This was previously allowed, but it was not

>   well-defined how to deal with placeholder parameters in this case.

Reviewer: Not true? At least, the spec currently says observable IDs can exist in observable formula but circular dependencies must be avoided. Although I don't really understand this -- typo?

Author: Thanks, I forgot to remove that sentence down there. It would work for observables without placeholder parameters, but I am not sure if that feature was important enough to add an exception there.

Reviewer: Fine for me to exclude :+1:

- ``observableParameters`` [NUMERIC, STRING OR NULL, OPTIONAL]

  Measurement-specific overrides for placeholder parameters declared in the
  observation model.

  The :ref:`observables table <v2_observables_table>` allows marking some
  parameters as measurement-specific (see below). Their values for a given
  measurement are specified in this column. The values are separated by

>   semicolons. The order (and number) of values must match the order (and number) of

Reviewer 2: ```suggestion
  semicolons. The order and number of values must match the order and number of
```

Reviewer 2: Do not see a need for parenthesis here.

  placeholders in the observable formula. The number of values must match

>   the number of placeholders in the observable formula. The values may be

Author: > seems fine to me to limit user error, but unnecessary.

Right. I'd find it more helpful to consider unused placeholders an error, but fine to change it if anybody considers that overly restrictive.

* ``observableFormula`` [STRING]

  Observation function as plain text formula expression.

  The expression may contain any symbol defined in the model,
  the mapping table or the parameter table.
  Often, this is just the ID of a state variable.
  Additionally, any observable ID introduced in the observable table
  may be referenced, but circular definitions must be avoided.
  Furthermore, any parameters introduced through the ``observablePlaceholders``
  field for the given observable may be used (see below).

* ``observablePlaceholders`` [STRING, OPTIONAL]

  A semicolon-separated list of valid PEtab identifiers that have not been
  introduced elsewhere, marking them as placeholders for
  measurement-specific parameters.
  The actual values for these parameters are specified in the
  ``observableParameters`` field of the measurement table.

>   The ordering of the symbols in ``observableParameters`` must match the
>   ordering of the symbols in ``observablePlaceholders``.

Reviewer: ```suggestion
  The ordering and number of values in ``observableParameters`` must match the
  ordering and number of placeholders in ``observablePlaceholders``.
```

  observable.

  Alternatively, some formula expression can be provided to specify
  more complex noise models. The same rules as for ``observableFormula``
  apply here. In addition, the current observable ID may be used to refer to
  the ``observableFormula`` expression.
  In addition to the placeholders declared in ``observablePlaceholders``,
  placeholders for noise parameters that are declared in ``noisePlaceholders``
  may be used.

> * ``noisePlaceholders`` [STRING, OPTIONAL]

Reviewer: ```suggestion
* ``noisePlaceholders`` [LIST[PETAB_ID], OPTIONAL]
```

>   A semicolon-separated list of valid PEtab identifiers that have not been

Reviewer: ```suggestion
  A semicolon-delimited list of valid PEtab identifiers that have not been
```

  introduced elsewhere, marking them as placeholders for
  measurement-specific noise parameters.
  The actual values for these parameters are specified in the
  ``noiseParameters`` field of the measurement table.

>   The ordering of the symbols in ``noiseParameters`` must match the
>   ordering of the symbols in ``noisePlaceholders``.

Reviewer: ```suggestion
  The ordering and number of values in ``noiseParameters`` must match the
  ordering and number of placeholders in ``noisePlaceholders``.
```

  For example, a noise model that accounts for measurement-specific relative
  and absolute contributions for some observable with the ID
  ``observable_pErk`` could be defined as::

    noise_offset_pErk + noise_scale_pErk * observable_pErk

  with ``noise_offset_pErk`` denoting the absolute and
  ``noise_scale_pErk`` the relative contribution.
  The corresponding ``noisePlaceholders`` would be::

    noise_offset_pErk;noise_scale_pErk

  and the ``noiseParameters`` in the measurement table could be, for example,
  ``0.1;0.2`` resulting in::

    noise_offset_pErk = 0.1

>     noise_scale_pErk = 0.2

Reviewer: :+1: 

Can refer to this as an example for observablePlaceholders too
